Add higher tiered fluid regulators

kstvr32 opened the pull request:
With changes from @Glease (GTNewHorizons/GT5-Unofficial#890), fluid regulators are no longer limited to storing their data in a single `int`. This makes it easy to add higher tiered fluid regulators to match the pumps.

This can be particularly useful for XL steam turbines in later tiers.

## Adds `Fluid Regulator (UHV)` and `Fluid Regulator (UEV)` as usable covers

![Screenshot 2024-09-07 182736](https://github.com/user-attachments/assets/957f35d8-e44d-4dc5-9e8f-ef5ec89675a0)

### UHV
![uhv](https://github.com/user-attachments/assets/6adffc48-7a9d-4046-ac81-0f354f3375d9)
![uhv_crafting](https://github.com/user-attachments/assets/a2dabcda-dafa-44a2-abb2-f24a9a1f5e73)

### UEV
![uev](https://github.com/user-attachments/assets/30b20672-4c4f-41a9-a341-4baf3e07b6f6)
![uev_crafting](https://github.com/user-attachments/assets/f8c3253b-ba88-412d-97c9-75cfaa285e30)

## Adds `Fluid Regulator (UIV)`, `Fluid Regulator (UMV)`, `Fluid Regulator (UXV)` and, `Fluid Regulator (MAX)` as crafting components

These are not usable covers matching the behavior of other high tiered pumps/conveyers. Instead, they can be used as crafting components.

![top_tiers](https://github.com/user-attachments/assets/0e6eba39-89af-4206-8fc3-f6907978c761)
(UMV, UXV, and MAX are animated)

![uiv_crafting](https://github.com/user-attachments/assets/adf299ed-3303-4dda-8d0f-b8b43022173a)
![umv_crafting](https://github.com/user-attachments/assets/a9549626-eac4-4af8-a5be-ef6e1c9e5004)
![uxv_crafting](https://github.com/user-attachments/assets/c17cc866-57f3-4972-a4bc-e0d1ed8e5e1b)
MAX does not have a crafting recipe as there are no craftable MAX circuits or MAX pumps.

OneEyeMaker commented on src/main/java/gregtech/api/enums/ItemList.java:
```diff
@@ -445,6 +445,8 @@ public enum ItemList implements IItemContainer {
     FluidRegulator_LuV,
     FluidRegulator_ZPM,
     FluidRegulator_UV,
+    FluidRegulator_UHV,
+    FluidRegulator_UEV,
```
Why don't you add all tiers (up to and including MAX)? At least for the sake of consistency with other components. Or is there some limitation or balance concern?

kstvr32 replied:
Pumps pants UEV are just crafting components. They can’t be used as covers. 

I could add all of the regulators, leave them uncraftable and unusable.

kstvr32 replied:
Thinking about it more, we could add even higher tiered pumps and regulators due to the change with how data is stored. Not sure if there are balance concerns.

OneEyeMaker replied:
I think there should be regulators past UEV, at least as crafting components. If there are no balance concerns, then both pump and regulator covers can be registered for those tiers.

kstvr32 replied:
Added fluid regulators all the way to MAX tier.

chochem reviewed (changes requested):
this pr appears to be broken and crashes the nightly https://discord.com/channels/181078474394566657/603348502637969419/1281823994806276187

please test your changes next time

chochem commented on src/main/java/gregtech/common/items/IDMetaItem01.java:
```diff
@@ -423,7 +423,9 @@ public enum IDMetaItem01 {
-    McGuffium_239(765);
+    McGuffium_239(765),
+    FluidRegulator_UHV(766),
+    FluidRegulator_UEV(767);
```
yea those indices are definetly out of bounds

kstvr32 replied:
Can I move this to metaitem 2/3?

kstvr32 replied:
Moving to a free range near the beginning of item 1.

kstvr32 commented:
> this pr appears to be broken and crashes the nightly https://discord.com/channels/181078474394566657/603348502637969419/1281823994806276187
> 
> please test your changes next time

This ran fine on my local. I can check again this afternoon. 

I thought there was enough room given that the signed short can be 32,767 giving 767 non-generated items.

chochem commented:
> This ran fine on my local.

pretty sure thats impossible

boubou19 commented:
it's possible it worked in dev env, but not fullpack

chochem commented:
> it's possible it worked in dev env, but not fullpack

it wouldnt crash but the items still wouldnt be there

kstvr32 commented:
Sorry about that, I’ll fix this and implement @OneEyeMaker suggested changes. 

Do you have an recommendations on a free set of IDs I can move this to?

Dream-Master commented:
@kstvr32 please finish the pr add the missing tiers and test your pr first. After this you can undraft it and ask for a re review.

kstvr32 commented:
Fixed PR. Implemented UHV and UEV fluid regulators correctly.

Still need texture for UMV, UXV, and MAX.

@OneEyeMaker should I add recipes for UIV, UMV, UXV, and MAX if they are not used as covers and not as crafting components (yet)?

Dream-Master commented:
Sure add textures and recipes @kstvr32

kstvr32 commented:
Added textures and recipes, re-opening PR.
